src_fun/parser.cpp

```diff
@@ -458,10 +457,22 @@ static bool parse_expression_leaf(Token_Stream* stream, Block_Builder* builder,
         if (!block)
             return false;
 
-        Parsed_Expression* expr = add_expression(builder, is_run ? EXPRESSION_RUN : EXPRESSION_UNIT, start, stream->cursor - 1, out_expression);
+        Parsed_Expression* expr = add_expression(builder, EXPRESSION_UNIT, start, stream->cursor - 1, out_expression);
         if (is_import) expr->flags |= EXPRESSION_UNIT_IS_IMPORT;
         expr->parsed_block = block;
     }
+    else if (maybe_take_atom(stream, ATOM_RUN))
+    {
+        if (lookahead_atom(stream, ATOM_LEFT_BRACE, 0))
+            return Report(stream->ctx)
+                .intro(SEVERITY_ERROR, stream->cursor)
+                .message("'run' expects a unit expression, not a block.\n"
+                         "Try passing a unit type instead."_s)

```

Comment:
Suggest prefixing the block with a "unit" keyword.

Reply:
Already doing this (next line)